src/registry/ReadonlyRegistry.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  ChainMap,
+  ChainMetadata,
+  ChainName,
+  WarpCoreConfig,
+  WarpRouteDeployConfig,
+} from '@hyperlane-xyz/sdk';
+import { Logger } from '@hyperlane-xyz/utils';
+import {
+  ChainAddresses,
+  MaybePromise,
+  UpdateChainParams,
+  WarpDeployConfigMap,
+  WarpRouteConfigMap,
+  WarpRouteFilterParams,
+} from '../types.js';
+import {
+  AddWarpRouteConfigOptions,
+  IRegistry,
+  IRegistryWriteMethod,
+  RegistryContent,
+  RegistryType,
+} from './IRegistry.js';
+import { MergedRegistry } from './MergedRegistry.js';
+
+/**
+ * Wraps and delegates all read operations (getChains, getMetadata, getAddresses, etc.)
+ * to the underlying registry while blocking all write operations (see {@link IRegistryWriteMethod}).
+ *
+ * Use this registry when you want to ensure that no modifications can be made to the underlying
+ * registry, such as when exposing a registry to untrusted code or when enforcing immutability
+ * at runtime.
+ *
+ * @example
+ * ```typescript
+ * const githubRegistry = new GithubRegistry({ branch: 'main' });
+ * const readonly = new ReadonlyRegistry(githubRegistry);
+ *
+ * // Read operations work normally
+ * const chains = await readonly.getChains();
+ *
+ * // Write operations are blocked and logged
+ * await readonly.addChain(newChain); // Logs a message, does nothing
+ * ```
+ */
+export class ReadonlyRegistry implements IRegistry {
```

Comment:
```suggestion
export class ReadonlyRegistry implements IReadRegistry {
```
are you against doing this?

Reply:
If i do this we won't be able to use the `ReadonlyRegistry` registry in the same places we can use a normal `IRegistry`. For example, in the http registry if wanted to have a readonly instance, we would not be able to as its service code accepts an `IRegistry` implementor. 

see:

- https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/infra/scripts/http-registry.ts#L23
- https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/http-registry-server/HttpServer.ts#L32